test/hack/resource/pkg/resourcetypes/eni.go

```diff
@@ -154,3 +160,59 @@ func (e *ENI) Cleanup(ctx context.Context, ids []string) ([]string, error) {
 
 	return deleted, errs
 }
+
+// get-vpc-resource-controller-enis is to get leaked ENIs by the vpc-resource-controller
+// Issue: https://github.com/aws/karpenter-provider-aws/issues/5582
+func (e *ENI) getVpcResourceControllerENIs(ctx context.Context, expirationTime time.Time, excludedClusters []string) (ids []string, err error) {
+	var nextToken *string
+	for {
+		out, err := e.cloudFormationClient.DescribeStacks(ctx, &cloudformation.DescribeStacksInput{
+			NextToken: nextToken,
+		})
+		if err != nil {
+			fmt.Println("dfdsf", err)
+			return ids, err
+		}
+
+		stacks := lo.Reject(out.Stacks, func(s cloudformationtypes.Stack, _ int) bool {
+			return s.StackStatus == cloudformationtypes.StackStatusDeleteComplete ||
+				s.StackStatus == cloudformationtypes.StackStatusDeleteInProgress ||
+				lo.FromPtr(s.CreationTime).After(expirationTime)
+		})
+		for _, stack := range stacks {
+			clusterName, found := lo.Find(stack.Tags, func(tag cloudformationtypes.Tag) bool {
+				return *tag.Key == karpenterTestingTag
+			})
+			if found {
+				if slices.Contains(excludedClusters, lo.FromPtr(clusterName.Value)) {
```

Comment:
Separate: Do we have a TODO comment around these excluded clusters to ensure that we eventually move this to a tagging mechanism? Minimally, do we have something that's tracked on GH?